src/mqt/yaqs/core/data_structures/noise_model.py

```
@@ -71,125 +254,218 @@ def __init__(
+            msg = "One-site processes do not accept 'factors'."
+            raise ValueError(msg)
+        if user_matrix:
+            proc["matrix"] = _as_square_matrix(proc["matrix"], "Process matrix")
+        else:
+            proc["matrix"] = NoiseModel.get_operator(name)
+        return proc
 
-            # One-site: ensure matrix
-            if "matrix" not in proc:
+    @staticmethod
+    def _normalize_two_site_process(
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)